### PR TITLE
change the check assert in split batch 2d

### PR DESCRIPTION
**Pull request by Wesley-Jzy**
Dim shape check should be put before we really prepare to chunk it.
https://github.com/hpcaitech/ColossalAI/issues/669

### CODE REVIEW

**Review by ver217 — APPROVED** [review]

File: colossalai/nn/layer/parallel_2d/_operation.py [hunk 1]
```diff
@@ -738,12 +738,13 @@ def split_batch_2d(input_: Tensor, dim: int = 0) -> Tensor:
         :class:`torch.tensor`: The tensor has been split.
     """
     dim_size = input_.size(dim)
+    if dim_size <= 1:
```
Comment (kurisusnowdeng) [inline colossalai/nn/layer/parallel_2d/_operation.py:741]: Not really correct.

**Review by kurisusnowdeng — CHANGES_REQUESTED** [review]

File: colossalai/nn/layer/parallel_2d/_operation.py (hunk 1 above)
Comment (Wesley-Jzy) [reply colossalai/nn/layer/parallel_2d/_operation.py:741]: I see. You are right.

**Review by kurisusnowdeng — APPROVED** [review]
